internal/service/base/resource_notification_policy.go

```diff
@@ -720,9 +1064,197 @@ func (p *NotificationPolicyResourceModel) expand(ctx context.Context) (*manageme
 		data.SetCountryLimit(countryLimit)
 	}
 
+	if !p.CooldownConfiguration.IsNull() && !p.CooldownConfiguration.IsUnknown() {
+		var cooldownConfigPlan NotificationPolicyCooldownConfigurationResourceModel
+		diags.Append(p.CooldownConfiguration.As(ctx, &cooldownConfigPlan, basetypes.ObjectAsOptions{
+			UnhandledNullAsEmpty:    false,
+			UnhandledUnknownAsEmpty: false,
+		})...)
+		if diags.HasError() {
+			return nil, diags
+		}
+
+		cooldownConfig, d := expandCooldownConfiguration(ctx, &cooldownConfigPlan)
+		diags.Append(d...)
+		if diags.HasError() {
+			return nil, diags
+		}
+
+		data.SetCooldownConfiguration(*cooldownConfig)
+	}
+
+	if !p.ProviderConfiguration.IsNull() && !p.ProviderConfiguration.IsUnknown() {
+		var providerConfigPlan NotificationPolicyProviderConfigurationResourceModel
+		diags.Append(p.ProviderConfiguration.As(ctx, &providerConfigPlan, basetypes.ObjectAsOptions{
+			UnhandledNullAsEmpty:    false,
+			UnhandledUnknownAsEmpty: false,
+		})...)
+		if diags.HasError() {
+			return nil, diags
+		}
+
+		providerConfig, d := expandProviderConfiguration(ctx, &providerConfigPlan)
+		diags.Append(d...)
+		if diags.HasError() {
+			return nil, diags
+		}
+
+		data.SetProviderConfiguration(*providerConfig)
+	}
+
 	return data, diags
 }
 
+func expandCooldownConfiguration(ctx context.Context, plan *NotificationPolicyCooldownConfigurationResourceModel) (*management.NotificationsPolicyCooldownConfiguration, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	email, d := expandCooldownConfigurationMethod(ctx, plan.Email)
+	diags.Append(d...)
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	sms, d := expandCooldownConfigurationMethod(ctx, plan.Sms)
+	diags.Append(d...)
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	voice, d := expandCooldownConfigurationMethod(ctx, plan.Voice)
+	diags.Append(d...)
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	whatsApp, d := expandCooldownConfigurationMethod(ctx, plan.WhatsApp)
+	diags.Append(d...)
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	cooldownConfig := management.NewNotificationsPolicyCooldownConfiguration(*email, *sms, *voice, *whatsApp)
+
+	return cooldownConfig, diags
+}
+
+func expandCooldownConfigurationMethod(ctx context.Context, methodObj types.Object) (*management.NotificationsPolicyCooldownConfigurationMethod, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	var methodPlan NotificationPolicyCooldownConfigurationMethodResourceModel
+	diags.Append(methodObj.As(ctx, &methodPlan, basetypes.ObjectAsOptions{
+		UnhandledNullAsEmpty:    false,
+		UnhandledUnknownAsEmpty: false,
+	})...)
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	method := &management.NotificationsPolicyCooldownConfigurationMethod{
+		Enabled: methodPlan.Enabled.ValueBool(),
+	}
+
+	if methodPlan.Enabled.ValueBool() {
```

Comment:
I think it's probably better to not make this check on Enabled. With this check we end up with some subtle validation (you can only set periods and resend_limit if enabled is true) as a part of the expand logic, which could be error-prone if the way this endpoint is validated changes in the future. It'd be better to just make the normal Null and Unknown checks for each attribute and leave validation to schema/modifyplan/validateconfig.